includes/class-wc-payments-order-service.php

```diff
@@ -1270,16 +1270,16 @@ private function generate_dispute_closed_note( $dispute_id, $status ) {
 	/**
 	 * Composes url for dispute details page.
 	 *
-	 * @param string $dispute_id Dispute id.
+	 * @param string $charge_id Charge ID.
 	 *
-	 * @return string Dispute details page url.
+	 * @return string Transaction details page url.
 	 */
-	private function compose_dispute_url( $dispute_id ) {
+	private function compose_dispute_url( $charge_id ) {
 		return add_query_arg(
 			[
 				'page' => 'wc-admin',
-				'path' => '/payments/disputes/details',
-				'id'   => $dispute_id,
+				'path' => '/payments/transactions/details',
```

Comment:
I notice one of the URL from the dispute order note has `/` in the query string parameter value. Forward slash should be encoded to `%2F`.

https://github.com/Automattic/woocommerce-payments/assets/73803630/5fd71eef-dd88-43fd-97fc-635e3c8a6a48

To confirm `encodeURIComponent('/')` in Javascript would return `%2F`.

I was surprised [add_query_arg()](https://developer.wordpress.org/reference/functions/add_query_arg/) does not encode the parameter value but then I read that [it expects the parameter value is already encoded](https://developer.wordpress.org/reference/functions/add_query_arg/#:~:text=Values%20are%20expected%20to%20be%20encoded%20appropriately%20with%20urlencode()%20or%20rawurlencode().).

```suggestion
				'path' => '%2Fpayments%2Ftransactions%2Fdetails',
```

or

```suggestion
				'path' => urlencode( '/payments/transactions/details' ),
```

Reply:
I've tried this and it ends up with a malformed URL with `0.00000` inserted in place of `/`.

```/wp-admin/admin.php?page=wc-admin&path=0.000000payments0.000000transactions0.000000details&id=ch_3Ny06gC15a3WbJ1t14zIF3QB```

This URL is being processed by through `WC_Payments_Utils::esc_interpolated_html()`, so I'll see if there's another way to urlencode.

Reply:
Ow wow. Not sure what's going on there. I only tried it on WP Console.

<img width="1613" alt="Screenshot 2023-10-06 at 08 50 23" src="https://github.com/Automattic/woocommerce-payments/assets/73803630/9bd68d47-382d-4b07-95c1-2bf09edf5cff">

Let's do it in a separate PR.

Reply:
I've opened an issue for this: #7399.